### PR TITLE
Adds connect with lines analysis

### PR DESCRIPTION
**Pull request by javierarce**
This PR adds the connect-with lines analysis composed of the following analyses: 

- `line-to-single-point`
- `line-to-column`
- `line-source-to-target`
- `line-sequential`

It also closes this issue: #9330 (adding the onboarding screen for the analysis)

Some screenshots: 

screen shot 2016-08-26 at 16 52 16
screen shot 2016-08-26 at 16 52 24
screen shot 2016-08-26 at 16 52 32
screen shot 2016-08-26 at 16 52 59

Error message in case there aren't second geometries in the `target`
screen shot 2016-08-26 at 11 24 18

Result of a connect with lines analysis: 

screen shot 2016-08-25 at 15 00 25

CR: @xavijam

### CODE REVIEW

**Comment by javierarce** [comment]
I'll be adding specs before merging.

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/connect-with-lines-form-model.js [hunk 1]
```diff
@@ -0,0 +1,329 @@
+var _ = require('underscore');
+var BaseAnalysisFormModel = require('./base-analysis-form-model');
+var template = require('./connect-with-lines.tpl');
+var ColumnOptions = require('../column-options');
+
+var CONNECT_WITH_LINES = require('./connect-with-lines-types');
+
+var TYPE_TO_META_MAP = {};
+CONNECT_WITH_LINES.map(function (d) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/connect-with-lines-form-model.js]: Why do we do that?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/connect-with-lines-form-model.js [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+  _onTypeChanged: function () {
+    this._replaceAttrs();
+    this._setSchema();
+  },
+
+  _replaceAttrs: function () {
+    var attrs = this.parse(this.attributes);
+    this.clear({ silent: true });
+    this.set('type', attrs.type, { silent: true }); // re-set type to avoid change:type event to trigger again
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/connect-with-lines-form-model.js]: I'd replace previous 2 steps with:

```js
this.attributes = {
  type: attrs.type
};
```

**Comment by xavijam** [comment]
Code LGTM, let's add those tests and ready to 🚢 🇮🇹 .

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/connect-with-lines-form-model.js (hunk 1 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/connect-with-lines-form-model.js]: It's our way to have different types of analysis inside of a main one. We use this approach in other analyses.

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/connect-with-lines-form-model.js]: aham! 👍

**Comment by javierarce** [comment]
Frontend specs are passing, so merging this branch. I'll handle adding spec in another branch to avoid delaying this feature.

**Pull request merged**